execution/gethexec/express_lane_service.go

```diff
@@ -366,9 +369,19 @@ func (es *expressLaneService) validateExpressLaneTx(msg *timeboost.ExpressLaneSu
 	if msg.AuctionContractAddress != es.auctionContractAddr {
 		return errors.Wrapf(timeboost.ErrWrongAuctionContract, "msg auction contract address %s does not match sequencer auction contract address %s", msg.AuctionContractAddress, es.auctionContractAddr)
 	}
-	currentRound := timeboost.CurrentRound(es.initialTimestamp, es.roundDuration)
-	if msg.Round != currentRound {
-		return errors.Wrapf(timeboost.ErrBadRoundNumber, "express lane tx round %d does not match current round %d", msg.Round, currentRound)
+
+	for {
+		currentRound := timeboost.CurrentRound(es.initialTimestamp, es.roundDuration)
+		if msg.Round == currentRound {
+			break
+		}
+
+		if msg.Round == currentRound+1 &&
```

Comment:
you can have weird conditions where currentRound is 7, byt by the time you computed timeTillNextRound currentRound became 8 and timeTillNextRound is now a full round's time.
easiest option would be to first take a timeStamp (time.Now) and then do both computations against that timestamp.
I'd also consider having some "expressLAneParams" that holds initialTimeStamp and roundDuration and has methonds like: timeTillNextRound (which takes no params) and timeTillNextRoundAt (which takes timestamp as param)

Reply:
Fixed the race condition. Not sure if making the params an object with methods makes much difference in clarity so I haven't done it for now.